sql/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java

```diff
@@ -314,13 +315,15 @@ private WindowDefinition getWindowDefinition(Optional<Window> maybeWindow,
         WindowFrameDefinition windowFrameDefinition = WindowDefinition.RANGE_UNBOUNDED_PRECEDING_CURRENT_ROW;
         if (window.getWindowFrame().isPresent()) {
             WindowFrame windowFrame = window.getWindowFrame().get();
-            validateFrame(window, windowFrame);
+            WindowFrameBoundsValidator.validateFrame(window, windowFrame);
```

Comment:
I think the changes here should also no longer be necessary?

Reply:
there are two parts to validation, one covers the invariant that is independent from the bound value, the other is a post validation once the values are bound. These two syntax checks go together and I thought the ..validator package contained classes related to validation. I think the code reads well.

Reply:
My point is that I don't understand why this change is necessary.

Given that the code has been moved it is also difficult to see the diff. It would be good to always split formating or code structure changes from behavior changes, so that it is easier to see the actual change.

Doing "bound value" validation in the analyzer sounds wrong to me. Queries have to fail with a appropriate error message in any case, so we have to assume that the analyzer *does not* have access to bound values. So the validation needs to happen later (planner or execution layer).

Reply:
I agree, reverting.